### PR TITLE
OOP School Library: Set Up Associations

### PR DESCRIPTION
**Pull request by DammyShittu**
## In this milestone, I completed the following requirements;

- Create a class `Classroom` with the following:
    - `@label` instance variable, should be initialized in the constructor.
    - Setter and getter for `@label` 
    
- Create the `has-many/belongs-to` relationship between `Classroom and Student`. The following should be implemented:
    - Create the `has-many` side (a classroom has many students).
    - Create the `belongs-to` side (a student belongs to a classroom).
    - Make sure that when adding a student to a classroom it also sets the classroom for the student.
    - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
    
- Create a class `Book` with the following:
    - `@title` and `@author` instance variables, should be initialized in the constructor.
    - Setters and getters for instance variables.
    
- Create a class `Rental` with the following:
    - `@date` instance variable, should be initialized in the constructor.
    - Setter and getter for `@date`.
    
 - Create the `many-to-many` relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
    - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
    - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
    - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
    - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
    - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rentals(rent)
+    @rentals << rent
+  end
```
Comment (Dipeshtwis) [inline book.rb]: - [ ] Good job on setting up the associations :+1:. I would recommend adding Getters and Setters for `@rentals`. Also, in the `add_rental` method you have to create the new rental achieving the `has-many` association. [ Hint: `Rental.new(date, self, person)`] (where the `self` is book itself that should be sent).

File: person.rb [hunk 2]
```diff
@@ -24,6 +25,10 @@ def validate_name
     @name = @corrector.correct_name(@name)
   end
 
+  def add_rentals(rent)
+    @rentals << rent
+  end
```
Comment (Dipeshtwis) [inline person.rb]: - [ ] I would recommend adding Getters and Setters for `@rentals`. Also, in the `add_rental` method you have to create the new rental achieving the `has-many` association. [ Hint: `Rental.new(date, book, self)`] (where person itself should be sent).

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Rental
+  attr_accessor :date
+  attr_reader :person, :book
```
Comment (Dipeshtwis) [inline rental.rb]: - [ ] I would recommend adding `book` and `person` to the attribute accessor as well as you need the getter and setter both.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DammyShittu, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental.rb (hunk 3 above)
Comment (DammyShittu) [reply rental.rb]: Thank you for this, @Dipeshtwis.
 I have effected the change.
Cheers.

File: book.rb (hunk 1 above)
Comment (DammyShittu) [reply book.rb]: Thank you @Dipeshtwis  👋 

I have implemented the required changes.

I will proceed to request another review now.

Cheers 🎉

**Review by mikethreels — APPROVED** [review]
Hi Dammy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 :
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by DammyShittu** [comment]
Thank you for the review, @mikethreels 👍

**Pull request merged**